frontend/packages/ceph-storage-plugin/src/components/namespace-store/create-ns.tsx

```diff
@@ -0,0 +1,251 @@
+  CreateNamespaceStoreFormProps & HandlePromiseProps
+>((props) => {
+  const { t } = useTranslation();
+  const [nsName, setNsName] = React.useState('');
+  const [provider, setProvider] = React.useState(BC_PROVIDERS.AWS);
+  const [providerDataState, providerDataDispatch] = React.useReducer(
+    providerDataReducer,
+    initialState,
+  );
+
+  const handleNsNameTextInputChange = (strVal: string) => {
+    if (strVal.length <= 43) {
+      setNsName(strVal);
+    }
+  };
+
+  const {
+    cancel,
+    className,
+    close,
+    inProgress,
+    errorMessage,
+    handlePromise,
+    isPage,
+    csv,
+    namespace,
+  } = props;
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    /** Create a secret if secret ==='' */
+    let { secretName } = providerDataState;
+    const promises = [];
+    if (!secretName) {
+      secretName = nsName.concat('-secret');
+      const { secretKey, accessKey } = providerDataState;
+      const secretPayload = secretPayloadCreator(
+        provider,
+        namespace,
+        secretName,
+        accessKey,
+        secretKey,
+      );
+      providerDataDispatch({ type: 'setSecretName', value: secretName });
+      promises.push(k8sCreate(SecretModel, secretPayload));
+    }
+    /** Payload for ns */
+    const nsPayload: Payload = {
+      apiVersion: apiVersionForModel(NooBaaNamespaceStoreModel),
+      kind: NooBaaNamespaceStoreModel.kind,
+      metadata: {
+        namespace,
+        name: nsName,
+      },
+      spec: {
+        type: NOOBAA_TYPE_MAP[provider],
+        ssl: false,
+      },
+    };
+    if (externalProviders.includes(provider)) {
+      nsPayload.spec = {
+        ...nsPayload.spec,
+        [PROVIDERS_NOOBAA_MAP[provider]]: {
+          [BUCKET_LABEL_NOOBAA_MAP[provider]]: providerDataState.target,
+          secret: {
+            name: secretName,
+            namespace,
+          },
+        },
+      };
+    }
+    if (provider === BC_PROVIDERS.S3) {
+      // eslint-disable-next-line
+      nsPayload.spec['s3Compatible'] = {
+        // eslint-disable-next-line
+        ...nsPayload.spec['s3Compatible'],
+        endpoint: providerDataState.endpoint,
+      };
+    } else if (provider === BC_PROVIDERS.IBM) {
+      nsPayload.spec.ibmCos = { ...nsPayload.spec.ibmCos, endpoint: providerDataState.endpoint };
+    }
+    // Add region in the end
+    if (provider === BC_PROVIDERS.AWS) {
+      nsPayload.spec.awsS3 = { ...nsPayload.spec.awsS3, region: providerDataState.region };
+    }
+
+    promises.push(k8sCreate(NooBaaNamespaceStoreModel, nsPayload));
+    return handlePromise(Promise.all(promises), (resource) => {
+      const lastIndex = resource.length - 1;
+      if (isPage)
+        history.push(
+          `/k8s/ns/${namespace}/clusterserviceversions/${getName(csv)}/${referenceForModel(
+            NooBaaNamespaceStoreModel,
+          )}/${getName(resource[lastIndex])}`,
+        );
+      else close();
+    });
```

Comment:
Directly pass habdler depicting the behavior on redirection
```suggestion
    props.redirectTo()
```

Reply:
What I mean is you can pass a handler.
This will make component better for extensibility.
For page the prop will be `redirectHandler`:
```
redirectHandler = (resourceName)  => history.push(
          `/k8s/ns/${namespace}/clusterserviceversions/${getName(csv)}/${referenceForModel(
            NooBaaNamespaceStoreModel,
          )}/${resource)}`,
        );
```
For non page:

```
redirectHandler = () => close()
```

This is similar to what you are doing for `cancel` props passing a handler ( L233 )
```
 <Button onClick={cancel} variant="secondary">
            {t('ceph-storage-plugin~Cancel')}
 </Button>
```